mypy/nodes.py

```diff
@@ -515,7 +515,6 @@ def __init__(self) -> None:
         # Original, not semantically analyzed type (used for reprocessing)
         self.unanalyzed_type: mypy.types.ProperType | None = None
         # If method, reference to TypeInfo
-        # TODO: Type should be Optional[TypeInfo]
```

Comment:
This comment was added in 2018 in https://github.com/python/mypy/pull/4438.

FakeInfo was added in 2017, and the following comment seems to be at odds:
https://github.com/python/mypy/blob/fdc7914b2001b57fda72b33adb725989dc63fe8a/mypy/nodes.py#L3366-L3370